### PR TITLE
First pass python extension for libucl

### PR DESCRIPTION
**Pull request by grimreaper**
This implements a first-pass python extension for libucl
- setup.py is more portable and more complete
- it only supports reading, not writing
- it might be better implemented with cffi
- still doesn't use safe iterators

but in the mean time it seems to work!

### CODE REVIEW

File: python/ucl.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	}
+	else if (obj->type == UCL_FLOAT) {
+		return Py_BuildValue("d", ucl_object_todouble (obj));
+	}
+	else if (obj->type == UCL_STRING) {
+		return Py_BuildValue("s", ucl_object_tostring (obj));
+	}
+	else if (obj->type == UCL_BOOLEAN) {
+		// maybe used 'p' here?
+		return Py_BuildValue("s", ucl_object_tostring_forced (obj));
```
Comment (vstakhov) [inline python/ucl.c]: Isn't there a boolean type in Python?

File: python/ucl.c [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+static PyObject*
+_iterate_valid_ucl(ucl_object_t const * obj) {
+	const ucl_object_t *tmp;
+	ucl_object_iter_t it = NULL;
+
+	tmp = obj;
+
+	while ((obj = ucl_iterate_object (tmp, &it, false))) {
```
Comment (vstakhov) [inline python/ucl.c]: I'd recommend to use safe iterators here since they have a better understanding of implicit arrays.

File: python/units.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#!/opt/local/bin/python
```
Comment (vstakhov) [inline python/units.py]: Erm, something like `/usr/bin/env python` should be more portable here.

File: python/ucl.c (hunk 1 above)
Comment (jbergstroem) [inline python/ucl.c]: Perhaps `PyBool_FromLong`? https://docs.python.org/3/c-api/bool.html

Comment (grimreaper) [inline python/ucl.c]: I wasn't sure if python truthyness matched ucl thruthyness.  This is what the comment about 'p' is for.

File: python/ucl.c (hunk 2 above)
Comment (grimreaper) [inline python/ucl.c]: I'll switch

File: python/units.py (hunk 3 above)
Comment (grimreaper) [inline python/units.py]: yeah, I wrote this at night and quickly.  I really need to fix setup.py to do this correctly.

**Comment by grimreaper** [comment]
I just added a bunch more tests and a stub validation function.  Ideally I'd like to get the basic module merged before i finish the whole thing.  Its hard to review that much code.

**Pull request merged**